tests/RobotFramework/tests/mqtt/custom_sub_topics_tedge-mapper-c8y.robot

```diff
@@ -0,0 +1,29 @@
+*** Settings ***
+Documentation       Purpose of this test is to verify that tedge-mapper-c8y subscribes the topics that are configured as c8y.topics
+
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Suite Teardown      Get Logs
+
+Test Tags           theme:mqtt    theme:c8y
+
+
+*** Test Cases ***
+Publish events to subscribed topic
+    Execute Command    mosquitto_pub -t tedge/events/event-type -m '{"text": "Event"}'
+    Should Have MQTT Messages    c8y/s/us
+
+Publish measurements to unsubscribed topic
+    Execute Command    mosquitto_pub -t tedge/measurements -m '{"temperature": 10}'
```

Comment:
Using `tedge mqtt pub` is more flexible here because `tedge mqtt pub` automatically reads the tedge.toml configuration and will use the secured broker client settings if required. It would make the test more flexible to future changes to the MQTT broker settings (e.g. making it secure by default and disabling the localhost:1883 endpoint in favour of the 8883 port).

Reply:
I was wondering which one is preferred for test, and you explained this exactly! Thanks!